packages/terra-button/src/Button.jsx

```diff
@@ -308,9 +308,9 @@ class Button extends React.Component {
     );
 
     // ignore aria-label for button with decorative icons.
-    let ariaLabel = (iconType !== IconTypes.DECORATIVE) ? customProps['aria-label'] : undefined;
-    if ((isIconOnly || variant === 'utility') && iconType === IconTypes.INFORMATIVE) {
-      ariaLabel = ariaLabel || text;
+    let ariaLabel = customProps['aria-label'] ? customProps['aria-label'] : text;
+    if ((isIconOnly || variant === 'utility') && iconType === IconTypes.DECORATIVE) {
+      ariaLabel = undefined;
     }
```

Comment:
Let's update this condition to like below so that it avoids adding unexpected aria-label to components like it has added for terra-alert and other components
```js
    let ariaLabel = customProps['aria-label'];
    if (isIconOnly || variant === 'utility') {
      ariaLabel = (iconType === IconTypes.INFORMATIVE) ? ariaLabel || text : undefined;
    }
```